### DISCUSSION TITLE
Using timeline variable in anonymous functions in v8

### DISCUSSION THREAD

**Discussion by holgerMitterer** [jspsych/jsPsych#3402] [Q&A]
I am just preparing my first experiment in v8, and I often generate prompts for audio-keyboard responses with a mixture of static and dyamic texts, like in the following:

`  
var cloze_trial = {

    type: jsPsychAudioKeyboardResponse,
    stimulus: jsPsych.timelineVariable('sound'),
    choices: ['q', 'p'],
    prompt: function () { 
        var trialChoices = "<p>Q: ";
        var leftOption  = jsPsych.timelineVariable('option_1', true);
        var rightOption = jsPsych.timelineVariable('option_2', true);
        console.log(leftOption);
        trialChoices = trialChoices + leftOption + "............................P: " + rightOption  + "</p>" ;
        console.log(trialChoices);
        return(trialChoices)
    },
    response_ends_trial: true
};`

In v8, the prompt looks like this (the timeline seems to work, the sound is playing):

Q: [object Object]............................P: [object Object]

I re-migrated the experiment to v7.3.3 and there it seemed to work, I get:

Q: contanto............................P: contanta
(with these two words provided by the timeline.

How has the usge of timeline variables in anonymous functions changed between v7 and v8?

Kind regards,
Holger

**Comment by Shaobin-Jiang** [comment]
Use `evaluateTimelineVariable` in a function. See [here](https://www.jspsych.org/latest/overview/timeline/#using-timeline-variables-in-a-function).
